scene/resources/packed_scene.cpp

```diff
@@ -380,7 +384,33 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 				//if node was not part of instance, must set its name, parenthood and ownership
 				if (i > 0) {
 					if (parent) {
-						parent->_add_child_nocheck(node, snames[n.name]);
+						bool pending_add = true;
+#ifdef TOOLS_ENABLED
+						if (Engine::get_singleton()->is_editor_hint()) {
+							Node *existing = parent->_get_child_by_name(snames[n.name]);
+							if (existing) {
+								// There's already a node in the same parent with the same name.
+								// This means that somehow the node was added both to the scene being
+								// loaded and another one instantiated in the former, maybe because of
+								// manual editing, or a bug in scene saving, or a loophole in the workflow
+								// (with any of the bugs possibly already fixed).
+								// Bring consistency back by letting it be assigned a non-clashing name.
+								// This simple workaround at least avoids leaks and helps the user realize
+								// something awkward has happened.
+								if (instantiation_warn_notify) {
+									instantiation_warn_notify(vformat("An incoming node's name clashes with %s already in the scene (presumably, from a more nested instance).\n"
```

Comment:
The message should be enclosed in `RTR()` so it can be extracted and translated.

I believe it doesn't support line breaks so you'll have to keep it all on a single line so it's properly extracted as a single msgid:
```
instantiation_warn_notify(vformat(RTR("An incoming node's name clashes with \"%s\" already in the scene (presumably, from a more nested instance).\nThe less nested node has been automatically renamed to resolve the conflict. Please review the changes and re-save the scene."),
		ret_nodes[0]->get_path_to(existing)));
```

Reply:
Added the `RTR()`. However, I've kept the line break since there should be no issue with it; quoting https://en.cppreference.com/w/cpp/language/string_literal:
> String literals placed side-by-side are concatenated at translation phase 6 (after the preprocessor). That is, "Hello,"  " world!" yields the (single) string "Hello, world!".

Reply:
I've changed my mind. One thing is C++; another one is the string extractor tool, which I can't find so I verify if this is compatible or not. I will just follow your advice.

Reply:
Yes the string extraction is done via a Python script which parses line by line: https://github.com/godotengine/godot-editor-l10n/blob/main/scripts/extract_editor.py#L51

Reply:
Shouldn't this be `TTR` tho? While it is not in a `#ifdef TOOLS_ENABLED` it is in a `if (is_editor_hint)` so this is still a tools only string. (I may also just be missing something about the translations, in which case disregard this comment)

Reply:
I think it could be either, both source their translations from the editor translations in priority. I think as a rule we use `RTR` for all `scene` to prevent mistakes where `TTR` would end up used for strings used at runtime.

Reply:
I've changed it to `TTR` since in the latest iterations it's within `#ifdef TOOLS_ENABLED`.